### PR TITLE
Refactor/restore asyncio

### PR DESCRIPTION
**Pull request by doronz88**

### CODE REVIEW

File: pymobiledevice3/restore/restore.py [hunk 1]
```diff
@@ -1140,23 +1132,23 @@ def send_streamed_image_decryption_key(self, message: Mapping) -> None:
...
+        await service.aio_send_plist({
             'ResponseBody': response.content,
             'ResponseBodyDone': True,
             'ResponseHeaders': dict(response.headers),
             'ResponseStatus': response.status_code,
-        }, fmt=plistlib.FMT_BINARY))
+        })
 
-    def send_component(self, component, component_name=None):
+    async def send_component(self, component, component_name: Optional[str] = None) -> None:
```
Comment (netanelc305) [inline pymobiledevice3/restore/restore.py:1142]: type hint

File: pymobiledevice3/restore/restored_client.py [hunk 2]
```diff
@@ -32,37 +40,37 @@ def _get_or_verify_udid(udid=None):
...
+        return await self.service.aio_send_recv_plist({'Request': 'Reboot', 'Label': self.label})
 
-    def send(self, message):
-        self.service.send_plist(message)
+    async def send(self, message: Mapping) -> None:
+        await self.service.aio_send_plist(message)
 
-    def recv(self):
-        return self.service.recv_plist()
+    async def recv(self):
```
Comment (netanelc305) [inline pymobiledevice3/restore/restored_client.py:67]: return type?

File: pymobiledevice3/restore/restored_client.py [hunk 3]
```diff
@@ -32,37 +40,37 @@ def _get_or_verify_udid(udid=None):
...
             req['RestoreOptions'] = opts.to_dict()
 
         self.logger.debug(f'start_restore request: {req}')
 
-        return self.service.send_plist(req)
+        return await self.service.aio_send_plist(req)
 
-    def reboot(self):
-        return self.service.send_recv_plist({'Request': 'Reboot', 'Label': self.label})
+    async def reboot(self) -> None:
```
Comment (netanelc305) [inline pymobiledevice3/restore/restored_client.py:61]: returning None or plist?

**Review by netanelc305 — CHANGES_REQUESTED** [review]

File: pymobiledevice3/restore/restore.py (hunk 1 above)
Comment (doronz88) [reply pymobiledevice3/restore/restore.py:1142]: done

File: pymobiledevice3/restore/restored_client.py (hunk 2 above)
Comment (doronz88) [reply pymobiledevice3/restore/restored_client.py:67]: done

File: pymobiledevice3/restore/restored_client.py (hunk 3 above)
Comment (doronz88) [reply pymobiledevice3/restore/restored_client.py:61]: fixed

**Pull request merged**